app/controllers/posts_controller.rb

```diff
@@ -29,13 +29,11 @@ def update
   end
 
   def index
-    @posts = Post.all
+    @posts = Post.published
     if params[:query].present?
-      @posts = Post.search(params[:query]).order('views_count DESC , created_at DESC')
-    else
-      @posts = Post.published.order('views_count DESC , created_at DESC')
+      @posts = @posts.search(params[:query])
```

Comment:
@babiracka I don't know Ruby that much… Do u really need to override `@post`?

Instead of doing this: 
`@posts = @posts.search(params[:query])`

Is it possible to do just this?
`@posts.search(params[:query])`